### PR TITLE
AP-706: Whitelist email domains

### PR DESCRIPTION
**Pull request by guganabu**

### CODE REVIEW

File: app/components/organization-email-domain/component.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  addDomain;
+
+  @task(function* (domain) {
+    try {
+      yield domain.deleteRecord();
+      yield domain.save();
+      this.domains.removeObject(domain);
+      this.notify.success(this.intl.t('whitelistEmailDomain.deleted'));
+    } catch (err) {
+      this.notify.error(parseError(err, 'Problem while deleting domain'));
```
Comment (Praseetha-KR) [inline app/components/organization-email-domain/component.js:75]: `Problem while deleting domain` has to be translated as well

File: app/components/organization-email-domain/component.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      this.domains.pushObject(domainInfo);
+    } catch {
+      return;
+    }
+    return;
+  }
+
+  @task(function* () {
+    this.domains = yield this.store.findAll('organization-email-domain');
+    console.log(' this.domains', this.domains);
```
Comment (Praseetha-KR) [inline app/components/organization-email-domain/component.js:41]: console.log!

**Review by Praseetha-KR — CHANGES_REQUESTED** [review]

File: translations/en.json [hunk 3]
```diff
@@ -863,5 +863,13 @@
...
-  "yourPlan": "Your Plan"
+  "yourPlan": "Your Plan",
+  "whitelistEmailDomain": {
+    "title": "Email Domain Restriction",
+    "delete": "Delete Domain",
+    "subDesc": "Whitelist email domains which are allowed to be invited as organization members. If not set all domains will be allowed.",
+    "add": "Add Domain",
+    "added": "added successfully",
+    "deleted": "Domain Deleted"
+  }
```
[the remark this answers is not in the document]
Comment (guganabu) [reply translations/en.json]: yes, updated the same

File: app/components/organization-email-domain/component.js (hunk 1 above)
Comment (guganabu) [reply app/components/organization-email-domain/component.js]: done

File: app/components/organization-email-domain/component.js (hunk 2 above)
Comment (guganabu) [reply app/components/organization-email-domain/component.js:41]: removed

**Review by Praseetha-KR — APPROVED** [review]

**Pull request merged**